diff --git a/plugins/linux-pipewire/pipewire.c b/plugins/linux-pipewire/pipewire.c
index 2b98c0e..5f2bedf 100644
--- a/plugins/linux-pipewire/pipewire.c
+++ b/plugins/linux-pipewire/pipewire.c
@@ -968,7 +968,7 @@ static void on_param_changed_cb(void *user_data, uint32_t id, const struct spa_p
 		buffer_types |= 1 << SPA_DATA_DmaBuf;
 #if PW_CHECK_VERSION(1, 2, 0)
 		obs_enter_graphics();
-		supports_explicit_sync = gs_query_sync_capabilities();
+		supports_explicit_sync = getenv("OBS_DISABLE_WL_SYNC") ? false : gs_query_sync_capabilities();
 		obs_leave_graphics();
 #endif
 	}
